PresenceInsightsSDK/PresenceInsightsSDK/PIAdapter.swift

```diff
@@ -745,24 +770,23 @@ extension PIAdapter {
     */
     private func buildRequest(endpoint:String, method:String, body: NSData?) -> NSURLRequest {
         
-        if let url = NSURL(string: endpoint) {
-            
-            let request = NSMutableURLRequest(URL: url)
-            
-            request.HTTPMethod = method
-            request.addValue(_authorization, forHTTPHeaderField: _httpAuthorizationHeader)
-            request.addValue(_contentTypeJSON, forHTTPHeaderField: _httpContentTypeHeader)
-            
-            if let bodyData = body {
-                request.HTTPBody = bodyData
-            }
-            
-            return request
-        } else {
+        guard let url = NSURL(string: endpoint) else {
             printDebug("Invalid URL: \(endpoint)")
+            // TODO: Should we throw ??
+            fatalError("Shouldn't be there ")
```

Comment:
This would happen if the developer entered the hostname incorrectly, or we messed up when adding a new endpoint. I think debug would be sufficient, but if the url is incorrect, then PIAdapter is completely useless, so I see it as a fatal error as well. What is best practice here?